File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/EmailAddressHistory.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "email_addresses_history", schema = "public")
public class EmailAddressHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "email_address_history_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @NotNull
    @Column(name = "email_address", nullable = false, length = Integer.MAX_VALUE)
    private String emailAddress;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "created_date_time")
    private Timestamp createdDateTime;

}

File: umbrella-insurance-react-ui/src/models/users/emailAddressesHistory/v1/EmailAddressHistory.ts
import { User } from "../../v1/User"

export class EmailAddressHistory {
      id?: number
      user?: User
      emailAddress?: string
      createdDateTime?: string

      constructor(obj?:any) {
            if(obj) {
                if(obj?.id) {
                    this.id = obj.id;
                }
                if(obj?.user) {
                    this.user = new User(obj.user);
                }
                if(obj?.emailAddress) {
                    this.emailAddress = obj.emailAddress;
                }
                if(obj?.createdDateTime) {
                    this.createdDateTime = obj.createdDateTime;
                }
            }
        }
}